lib/livebook_web/helpers/codec.ex

```diff
@@ -0,0 +1,133 @@
+defmodule LivebookWeb.Helpers.Codec do
+  @wav_header_size 44
+
+  @doc """
+  Returns the size of a WAV binary that would wrap PCM data of the
+  given size.
+
+  This size matches the result of `encode_pcm_as_wav_stream!/6`.
+  """
+  @spec pcm_as_wav_size(pos_integer()) :: pos_integer()
+  def pcm_as_wav_size(pcm_size) do
+    @wav_header_size + pcm_size
+  end
+
+  @doc """
+  Encodes PCM float-32 in native endianness into a WAV binary.
+
+  Accepts a range of the WAV binary that should be returned. Returns
+  a stream, where the PCM binary is streamed from the given file.
+  """
+  @spec encode_pcm_as_wav_stream!(
+          Path.t(),
+          non_neg_integer(),
+          pos_integer(),
+          pos_integer(),
+          non_neg_integer(),
+          pos_integer()
+        ) :: Enumerable.t()
+  def encode_pcm_as_wav_stream!(path, file_size, num_channels, sampling_rate, offset, length) do
+    {header_enum, file_length} =
+      if offset < @wav_header_size do
+        header = encode_pcm_as_wav_header(file_size, num_channels, sampling_rate)
+        header_length = min(@wav_header_size - offset, length)
+        header_slice = binary_slice(header, offset, header_length)
+        {[header_slice], length - header_length}
+      else
+        {[], length}
+      end
+
+    file_offset = max(offset - @wav_header_size, 0)
+
+    file_stream = raw_file_range_stream!(path, file_offset, file_length)
+
+    file_stream =
+      case System.endianness() do
+        :little ->
+          file_stream
+
+        :big ->
+          Stream.map(file_stream, fn binary ->
+            for <<x::32-float-native <- binary>>, reduce: <<>> do
```

Comment:
For clarity?

```suggestion
            for <<x::32-float-big <- binary>>, reduce: <<>> do
```

Reply:
Wait, but that's not the same? The binary literally uses `System.endianness()`.

Reply:
It is the same, yeah. I think big is clearer but it is your call.